fractal_graphing/http/http.go

```diff
@@ -0,0 +1,108 @@
+// Copyright © 2016 Pawel Rozlach.
+// License: https://creativecommons.org/licenses/by-nc-sa/4.0/
+
+// Based on work by Alan A. A. Donovan & Brian W. Kernighan
+// which can be found at:
+// https://github.com/adonovan/gopl.io.git
+
+// Mandelbrot emits a PNG image of the Mandelbrot fractal.
+package http
+
+import (
+	"fmt"
+	"image/png"
+	"io"
+	"net/http"
+	"os"
+	"strconv"
+
+	"github.com/vespian/go-excersizes/fractal_graphing/algos"
+	"github.com/vespian/go-excersizes/fractal_graphing/common"
+	"github.com/vespian/go-excersizes/fractal_graphing/img"
+)
+
+func ServeHttp(w http.ResponseWriter, r *http.Request) {
+	width, height, scaling, algo, r_err := getHttpReqData(r)
+
+	if r_err != nil {
+		w.Header().Set("Content-Type", "text/plain")
+		msg := fmt.Sprintf("getHttpReqData failed: %v", r_err)
+		// (sur) How paranoid should I be ? do I need to do error checking
+		// for each and every command or just common sense ?
```

Comment:
err checks, err checks everywhere :-) yes, be paranoid about errors!
errors are just values and are treated equally like the happy case. errors are not "exceptional" as in some languages.